packages/playground/src/utils/validators.ts

```diff
@@ -777,3 +777,20 @@ export async function isAvailableName(grid: GridClient, name: string) {
     return { message: "Name is already taken." };
   }
 }
+
+/**
+ * Validates an SMTP input string.
+ *
+ * Checks if the input string is a valid email address and does not contain special characters.
+ *
+ * @param {string} input - The input string to validate.
+ * @returns {{ message: string }} - An object with an error message if the input is invalid, or undefined if the input is valid.
+ */
+
+export function isValidSmtp(input: string) {
+  const emailValidation = isEmail("Please provide a valid email address.")(input);
+  const username = /[!@#$%^&*()\s_+\-={}:<>?,./]/.test(input);
```

Comment:
This could have been simplified instead of `matching against special characters` you could have just made it `match against valid characters` which are `a-z[A-Z0-9]+`

Reply:
> This could have been simplified instead of `matching against special characters` you could have just made it `match against valid characters` which are `a-z[A-Z0-9]+`

Please note this is the right way, given there's a very long of list of unhandled utf8 input, that's why you should state the allowed ones as long as you can iterate them, instead of chasing a very long list of symbols.

Reply:
thanks, the regex was implemented separately in the validator function and used in the input validation.